api-design-guidelines/data-modelling.md

```diff
@@ -0,0 +1,275 @@
+---
+id: esTmApiGuidelinesDataModelling
+slug: /elasticsearch-team/api-guidelines/data-modelling
+title: Data modelling
+image: https://source.unsplash.com/400x175/?Nature
+description: API guidelines - Data modelling
+date: 2022-06-26
+tags: ['elasticsearch','internal docs']
+---
+
+Request and response bodies are typically encoded in JSON. The full JSON specification is defined in [RFC 8259](https://datatracker.ietf.org/doc/html/rfc8259).
+
+## JSON objects are not ordered
+
+JSON objects must be conceptually treated as **unordered**. Maintaining order of deserialized JSON objects is not supported in general, and isn’t possible in some programming languages.
+
+If preservation of order is important, such ordering information should be represented separately. The examples below shows a plain (unordered) object alongside three alternative representations which also provide ordering detail.
+
+```json
```

Comment:
The comments inside the JSON code block are highlighted as "invalid syntax". Let's maybe use this little hack to make it a little bit prettier (`jsonc` or `json5` could as well be used, but that does not highlight the field names in the same way as `json`):

``````suggestion
```yaml
``````

Reply:
I agree that this is "broken" but I have ported over the files as-is from their source, where this invalid JSON also exists. I'm wondering whether this should be fixed here or in a separate commit later.